cilium-cli/install/upgrade.go

```diff
@@ -27,14 +28,33 @@ func (k *K8sInstaller) UpgradeWithHelm(ctx context.Context, k8sClient *k8s.Clien
 	if err != nil {
 		return err
 	}
+	
+	// Determine the effective behavior based on flags
+	var existingValues map[string]interface{}
+	if !k.params.HelmResetValues && !k.params.HelmReuseValues {
```

Comment:
I'm not sure I'm understanding the logic here. Why are you not simply setting the `ResetThenReuseValues` option of the helm client, instead of implementing a custom version?

Reply:
Sorry, now I set ResetThenReuseValues. At first, I thought it would be better to implement a custom one.